### PR TITLE
Add "Close" and "Learn More" buttons to TensorBoard.dev Upload dialog

### PR DESCRIPTION
**Pull request by bmd3k**
* Motivation for features / changes
In the "Upload to TensorBoard.dev" dialog we want to add a couple buttons that allow the user to "Close" the dialog or "Learn More" about TensorBoard.dev.

* Technical description of changes
Add a couple material buttons at the bottom of tbdev_upload_dialog_component.

* Screenshots of UI changes
image

* Detailed steps to verify changes work correctly (as executed by you)
Patched in changes that integrate the Upload button into the header.  Started up a server. Clicked on "Upload". Interacted with buttons and verified that "Close" closes the dialog and "Learn More" opens a new tab for TensorBoard.dev.

### CODE REVIEW

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html [hunk 1]
```diff
@@ -39,3 +39,16 @@ <h3>Upload to TensorBoard.dev?</h3>
...
 </p>
+<div class="bottom-buttons">
+  <button mat-flat-button class="close-button" (click)="onClose()">
+    Close
+  </button>
+  <a
+    mat-flat-button
+    class="learn-more-button"
+    href="https://tensorboard.dev"
+    target="_blank"
```
Comment (psybuzz) [inline tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: For <a> tags, please add `rel="noopener noreferrer"`, similar to other anchor tags.

File: tensorboard/webapp/tbdev_upload/tbdev_upload_test.ts [hunk 2]
```diff
@@ -76,4 +80,16 @@ describe('tbdev upload test', () => {
...
   });
+
+  it('can be closed with button', async () => {
+    const fixture = TestBed.createComponent(TbdevUploadDialogComponent);
+    fixture.detectChanges();
+
+    const copyElement = fixture.debugElement.query(By.css('.close-button'));
+    copyElement.nativeElement.click();
+    fixture.detectChanges();
+    await fixture.whenStable();
```
Comment (psybuzz) [inline tensorboard/webapp/tbdev_upload/tbdev_upload_test.ts:91]: I'm not 100% sure, but is this `await whenStable` required?  If it is, I'm curious to learn more about when it is needed

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss [hunk 3]
```diff
@@ -52,3 +52,20 @@ code {
...
 }
+
+.bottom-buttons {
+  align-items: center;
+  display: flex;
+  justify-content: flex-end;
+}
+
+.close-button {
+  color: map-get($tb-foreground, secondary);
```
Comment (psybuzz) [inline tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss:63]: I'm guessing this is supposed to be `secondary-text` not `secondary`?

from `tensorboard/webapp/theme/_tb_theme.template.scss`

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html (hunk 1 above)
Comment (bmd3k) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: I added rel="noopener" but I would like to pass referrer information for tensorboard.dev analytics. Would that work?

File: tensorboard/webapp/tbdev_upload/tbdev_upload_test.ts (hunk 2 above)
Comment (bmd3k) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_test.ts]: Done. Indeed we did not need these two calls to fixture.

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss (hunk 3 above)
Comment (bmd3k) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.scss]: Done. Good catch.

File: tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html (hunk 1 above)
Comment (psybuzz) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: Ack, only `rel="noopener"` sounds fine for this case.

Comment (wchargin) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: Please hold on this pending internal discussion.

Comment (bmd3k) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: After internal discussion we decided to include noreferrer.

Comment (wchargin) [reply tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ng.html:50]: Thanks!

**Pull request merged**